### PR TITLE
add changelog for v0.5.0 release

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- add changelog for v0.5.0 release

#### Ticket Link
n/a

#### Release Note

```release-note
add changelog for v0.5.0 release
```

### CODE REVIEW

**Review by dlorenc — APPROVED** [review]

**Comment by dlorenc** [comment]
I think we're ready for this one today!

**Review by asraa — APPROVED** [review]
lgtm! should we include the fix to verify minisign sigs from the latest minisign version v.10?

**Comment by cpanato** [comment]
> lgtm! should we include the fix to verify minisign sigs from the latest minisign version v.10?

which pr was that?

**Comment by asraa** [comment]
> which pr was that?


https://github.com/sigstore/rekor/pull/639

**Comment by cpanato** [comment]
added @asraa PTAL

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,77 @@
...
+* Bump golang from `8c0269d` to `0fa6504` (https://github.com/sigstore/rekor/pull/597)
+* Pin dependencies in github action workflows and Dockerfile (https://github.com/sigstore/rekor/pull/595)
+* update release image to use go 1.17.6 (https://github.com/sigstore/rekor/pull/589)
+* Bump golang from 1.17.5 to 1.17.6 (https://github.com/sigstore/rekor/pull/588)
+* Bump go.uber.org/goleak from 1.1.11 to 1.1.12 (https://github.com/sigstore/rekor/pull/585)
+* Bump go.uber.org/zap from 1.19.1 to 1.20.0 (https://github.com/sigstore/rekor/pull/584)
+* Bump github.com/go-playground/validator/v10 from 10.9.0 to 10.10.0 (https://github.com/sigstore/rekor/pull/579)
+* Bump actions/github-script from 4 to 5 (https://github.com/sigstore/rekor/pull/577)
+
+## Contributors
```
Comment (asraa) [inline CHANGELOG.md:64]: i will request myself here :)

**Review by asraa — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (cpanato) [reply CHANGELOG.md:64]: OOOHHH so sorry, i thought i added but looks like i overlook 
i'm ashamed :(

Comment (cpanato) [reply CHANGELOG.md:64]: added, my apologizes again

Comment (asraa) [reply CHANGELOG.md:64]: hahahah no worries at alll!!!

**Review by asraa — APPROVED** [review]
Thank you Carlos!!

**Pull request merged**